nipype/interfaces/base/traits_extension.py

```diff
@@ -54,244 +95,239 @@ class Str(Unicode):
 traits.DictStrStr = DictStrStr
 
 
-class File(BaseUnicode):
-    """ Defines a trait whose value must be the name of a file.
-    """
+class BasePath(TraitType):
+    """Defines a trait whose value must be a valid filesystem path."""
 
     # A description of the type of value this trait accepts:
-    info_text = 'a file name'
-
-    def __init__(self,
-                 value='',
-                 filter=None,
-                 auto_set=False,
-                 entries=0,
-                 exists=False,
-                 **metadata):
-        """ Creates a File trait.
-
-        Parameters
-        ----------
-        value : string
-            The default value for the trait
-        filter : string
-            A wildcard string to filter filenames in the file dialog box used by
-            the attribute trait editor.
-        auto_set : boolean
-            Indicates whether the file editor updates the trait value after
-            every key stroke.
-        exists : boolean
-            Indicates whether the trait value must be an existing file or
-            not.
-
-        Default Value
-        -------------
-        *value* or ''
-        """
-        self.filter = filter
-        self.auto_set = auto_set
-        self.entries = entries
+    info_text = 'a pathlike object or string'
+    exists = False
+    pathlike = False
+    resolve = False
+    _is_file = False
+    _is_dir = False
+
+    def __init__(self, value=Undefined,
+                 exists=False, pathlike=False, resolve=False, **metadata):
+        """Create a BasePath trait."""
         self.exists = exists
+        self.resolve = resolve
+        self.pathlike = pathlike
+        if any((exists, self._is_file, self._is_dir)):
+            self.info_text += ' representing a'
+            if exists:
+                self.info_text += 'n existing'
+            if self._is_file:
+                self.info_text += ' file'
+            elif self._is_dir:
+                self.info_text += ' directory'
+            else:
+                self.info_text += ' file or directory'
```

Comment:
Does it make sense to make `info_text` a property, instead? I realize basically nobody plays with traits directly, but:

```Python
>>> x = File(exists=True)
>>> x.info_text
'a pathlike object or string representing an existing file'
>>> x.exists = False
>>> x.info_text
'a pathlike object or string representing an existing file'
```

This probably doesn't matter.

Reply:
I believe we don't change the `exists` metadata of inputs/outputs but I'll give it a try.

Reply:
Yeah, we don't. It's just a general feeling that it's still a Python object, and people could modify it, and maybe we should behave "correctly" in that case.

Again, it probably doesn't matter because almost certainly nobody does use our API that way. Just couldn't help myself commenting.